run.py

```diff
@@ -0,0 +1,163 @@
+    summaryMetrics : dict
+        dict of summary statistics for the run. E.g., `{'loss': 0.5, 'accuracy': 0.9}`.
+    history : list of dict
+        Iterable of dicts containing the arguments to calls of wandb.log
+        made during the run. E.g., [{"loss": 10}, {"a": 9}, {"a": 8}, {"a": 7}]
+    search_info: dict
+        For runs in the proposed state, information produced by the optimizer. E.g.,
+        {'improvement_prob': 0.2}
+    early_terminate_info: dict
+        For runs in the proposed state, information produced by the optimizer. E.g.,
+        {'improvement_prob': 0.2}
+    """
+
+    name: Optional[str] = None
+    summary_metrics: dict = Field(default_factory=lambda: {})
+    history: List[dict] = Field(default_factory=lambda: [])
+    config: dict = Field(default_factory=lambda: {})
+    state: RunState = RunState.proposed
+    search_info: Optional[Dict] = None
+    early_terminate_info: Optional[Dict] = None
+
+    class Config:
+        use_enum_values = True
+
+    def metric_history(self, metric_name: str) -> List[floating]:
+        return [d[metric_name] for d in self.history if metric_name in d]
+
+    def summary_metric(self, metric_name: str) -> floating:
+        if metric_name not in self.summary_metrics:
+            raise KeyError(f"{metric_name} is not a summary metric of this run.")
+        return self.summary_metrics[metric_name]
+
+    def metric_extremum(self, metric_name: str, kind: str) -> floating:
+        """Extract the value of the target optimization metric from a
+         specified sweep run.
+
+        Parameters
+        ----------
+        sweep_config: SweepConfig or dict
+            The sweep configuration, where the name of the target metric
+            is specified.
+        run: SweepRun
+            The run to extract the value of the metric from.
+        default: float, optional, default None
+            The default value to use if no metric is found.
+
+        Returns
+        -------
+        metric: float
+            The run's metric.
+        """
+
+        cmp_func = np.max if kind == "maximum" else np.min
+        try:
+            summary_metric = [self.summary_metric(metric_name)]
+        except KeyError:
+            summary_metric = []
+        all_metrics = self.metric_history(metric_name) + summary_metric
+
+        if len(all_metrics) == 0:
+            raise ValueError(f"Cannot extract metric {metric_name} from run")
+
+        def filter_func(x: Any) -> bool:
+            try:
+                return np.isscalar(x) and np.isfinite(x)
+            except TypeError:
+                return False
+
+        all_metrics = list(filter(filter_func, all_metrics))
+
+        if len(all_metrics) == 0:
+            raise ValueError("Run does not have any finite metric values")
+
+        return cmp_func(all_metrics)
+
+
+def next_run(
+    sweep_config: Union[dict, SweepConfig], runs: List[SweepRun], **kwargs
```

Comment:
yea kwargs would be passed in through anaconda, in principle. I removed it though as it is not currently used anywhere or in anaconda right now. Really, we should make all options and search parameters be specified in the sweep config rather than to python directly.

added a bunch of examples to `README.md`.